var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -92,6 +117,15 @@ def setup_dependent_package(self, module, dependent_spec):
             join_path(self.prefix.lib, 'libmpi.{0}'.format(dso_suffix))
         ]
 
+    def autoreconf(self, spec, prefix):
+        """Not needed usually, configure should be already there"""
+        # If configure exists nothing needs to be done
+        if os.path.exists(self.configure_abs_path):
+            return
+        # Else bootstrap with autotools
+        bash = which('bash')
+        bash('./autogen.sh')
```

Comment:
Was the default `autoreconf` failing?

Reply:
Forgive my ignorance, but does the default `autoreconf` run `autogen.sh` if it is found in the source directory? Simply running `autoreconf -vif` will not completely bootstrap MPICH. There are other Perl scripts run by `autogen.sh` that generate things like the language bindings.